services/src/northbound/src/main/java/org/openkilda/northbound/messaging/kafka/KafkaMessageConsumer.java

```diff
@@ -122,36 +94,8 @@ public Message poll(final String correlationId) {
                 OPERATION_TIMED_OUT, TIMEOUT_ERROR_MESSAGE, northboundTopic);
     }
 
-    //todo(Nikita C): rewrite current poll method using async way.
-    /*
-    @Async
-    public CompletableFuture<Message> asyncPoll(final String correlationId) {
-        try {
-            for (int i = POLL_TIMEOUT / POLL_PAUSE; i < POLL_TIMEOUT; i += POLL_PAUSE) {
-                if (messages.containsKey(correlationId)) {
-                    return CompletableFuture.completedFuture(messages.remove(correlationId));
-                } else if (messages.containsKey(SYSTEM_CORRELATION_ID)) {
-                    return CompletableFuture.completedFuture(messages.remove(SYSTEM_CORRELATION_ID));
-                }
-                Thread.sleep(POLL_PAUSE);
-            }
-        } catch (InterruptedException exception) {
-            logger.error("{}: {}={}", INTERRUPTED_ERROR_MESSAGE, CORRELATION_ID, correlationId);
-            throw new MessageException(correlationId, System.currentTimeMillis(),
-                    INTERNAL_ERROR, INTERRUPTED_ERROR_MESSAGE, Topic.NORTHBOUND);
-        }
-        logger.error("{}: {}={}", TIMEOUT_ERROR_MESSAGE, CORRELATION_ID, correlationId);
-        throw new MessageException(correlationId, System.currentTimeMillis(),
-                OPERATION_TIMED_OUT, TIMEOUT_ERROR_MESSAGE, Topic.NORTHBOUND);
-    }
-    */
-
-    /**
-     * {@inheritDoc}
-     */
     @Override
-    public void clear() {
-        //we shouldn't clear up collection, outdated messages are removing by default.
-        //messages.clear();
+    public void onResponse(Message message) {
```

Comment:
Is it thread safe?

Reply:
KafkaMessageConsumer.java will be deleted